### ISSUE TITLE
[Bug]: Action Button Accessibility could be improved

### ISSUE THREAD

**Issue by adixon-adobe** [adobe/spectrum-web-components#2791]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-action-button

### Expected behavior

Easier and clearer accessibility patterns for action button text.

I think ideally there would be both a `label` attribute as well as either a `compact` or `hideLabel` attribute to control it's visibility (or some other similar pattern that allows putting the text in one place, and selectively showing it in the UI).

### Actual behavior

The documentation doesn't make what to do clear, and from an internal slack discussion it looks like sp-action-button may not be adding some aria tags it really should with the information it has.

I'm writing some code where I'd like to selectively put the action button in a compact mode on smaller screens when the text doesn't fit (in which case the buttons should be icon-only). Ideally I can share code between the 2 modes, and keep the `compact` logic to a single attribute.

**Comment by Westbrook** [comment]
First, to clarify, external conversation around appropriate attributes being applied was based on user error, which certainly points to the benefit of updated documentation here. You can review successfully accessible consumption of this patter [here](https://studio.webcomponents.dev/edit/mOF1zUEjLJzODGXFYaIU/src/index.ts?branch=no-visible-label%40wfVpXdsTYhf9mY2slxW4t2Y2SjC3&p=stories) while we discuss the correct documentation path and any changes that might be necessary to the actual element API.

For all elements, I prefer an HTML first content API, with options of attribute/property based APIs that do not disturb that usage. That is to say that `<sp-action-button>Label text</sp-action-button>` is always preferred to `<sp-action-button label="Label text"></sp-action-button>`, even if this consumption isn't quite even across the library today. With that in mind, actual API additions to support something like this would need to take into account label application across the library, including usage in components like `<sp-slider>` which supports a `labelVisibility?: 'text' | 'value' | 'none';` attribute/property and `<sp-dialog-wrapper>` which support a `headlineVisibility: 'none' | undefined;` attribute property, along with other usage of the `label` attribute property directly. 

Related, it would be good to update the consumption of `label` in the "button" family of elements so that `aria-label` was removed when it was not present, not simply emptied, see: https://github.com/adobe/spectrum-web-components/blob/sp-update-overlays/packages/button/src/ButtonBase.ts#L231

**Comment by castastrophe** [comment]
We have this feedback tracked in our Jira epic around accessibility documentation so I'll close this issue. Please let us know if you need a public-facing update!
